src/hotspot/share/classfile/placeholders.hpp

```diff
@@ -115,10 +117,10 @@ class PlaceholderEntry {
   Symbol*            supername()           const { return _supername; }
   void               set_supername(Symbol* supername);
 
-  JavaThread*        definer()             const {return _definer; }
-  void               set_definer(JavaThread* definer) { _definer = definer; }
+  JavaThread*        definer_acquire()     const { return Atomic::load_acquire(&_definer); }
+  void               release_set_definer(JavaThread* definer) { Atomic::release_store(&_definer, definer); }
```

Comment:
I cannot see any lock-free access to the definer thread that would require these acquire/release semantics. AFAICS all reading and writing of the definer happen under the SD lock - even the `print_on` usage is ultimately under the SD lock.

Reply:
You're right.  I thought this access required an acquire but it is always after you relock the SystemDictionary_lock, so it will refetch the field from probe ?

```
    while (probe->definer_acquire() != nullptr) {
      SystemDictionary_lock->wait();
    }
```

Reply:
Yes.